internal/internal_task_handlers_test.go

```diff
@@ -1306,19 +1306,34 @@ func (t *TaskHandlersTestSuite) TestLocalActivityRetry_WorkflowTaskHeartbeatFail
 func (t *TaskHandlersTestSuite) TestHeartBeat_NoError() {
 	mockCtrl := gomock.NewController(t.T())
 	mockService := workflowservicemock.NewMockWorkflowServiceClient(mockCtrl)
-
+	invocationChannel := make(chan int, 2)
 	heartbeatResponse := workflowservice.RecordActivityTaskHeartbeatResponse{CancelRequested: false}
-	mockService.EXPECT().RecordActivityTaskHeartbeat(gomock.Any(), gomock.Any(), gomock.Any()).Return(&heartbeatResponse, nil)
+	mockService.EXPECT().
+		RecordActivityTaskHeartbeat(gomock.Any(), gomock.Any(), gomock.Any()).
+		Do(func(_ interface{}, _ interface{}, _ ...interface{}) { invocationChannel <- 1 }).
+		Return(&heartbeatResponse, nil).
+		Times(2)
 
 	temporalInvoker := &temporalInvoker{
-		identity:  "Test_Temporal_Invoker",
-		service:   mockService,
-		taskToken: nil,
+		identity:         "Test_Temporal_Invoker",
+		service:          mockService,
+		taskToken:        nil,
+		heartBeatTimeout: time.Second,
 	}
 
 	heartbeatErr := temporalInvoker.Heartbeat(nil, false)
+	t.NoError(heartbeatErr)
 
+	heartbeatErr = temporalInvoker.Heartbeat(nil, false)
 	t.NoError(heartbeatErr)
+
```

Comment:
assert that there is no invocation directly after the 2nd heartbeat call returns. this guarantees the 2nd invocation comes via the heartbeat callback